### ISSUE TITLE
[primTorch] Feature Request: out-of-place fill

### ISSUE THREAD

**Issue by mruberry** [pytorch/pytorch#77932]
Currently fill only has an inplace version, but it's a natural prim.

cc @ezyang @mruberry @ngimel

**Comment by ngimel** [comment]
Is out of place `fill`=`full_like`?

**Comment by mruberry** [comment]
> Is out of place `fill`=`full_like`?

Yes except that full_like won't setup a gradient relationship between the input and output tensors.

**Comment by mruberry** [comment]
A very ugly implementation:

```
def fill(a, value):
    t = a * False
    with torch.no_grad():
        t.fill_(value)
    return t

**Comment by ezyang** [comment]
Cc @bdhirsh

**Comment by peterbell10** [comment]
`torch.fill` was added in #76084.
